### PR TITLE
use btree_map for storing withdrawals mapped to the account

### PR DESCRIPTION
**Pull request by felixfaisal**
Using a BTreeMap for storing withdrawals

### CODE REVIEW

**Comment by serhii-temchenko** [comment]
Had discussion on a standup about this change with @35359595, please contact him about that.

**Comment by serhii-temchenko** [comment]
If discussed improvement with tests of serialization is in progress - please remove `review` label.

File: src/snapshot.rs [hunk 1]
```diff
@@ -63,4 +64,39 @@ impl<Account: std::cmp::Ord, Balance: Zero + Clone, WithdrawalLimit: Get<u32>, A
...
+mod test{
+    use sp_runtime::AccountId32;
+    use std::convert::TryInto;
+    use std::convert::From;
+    use std::collections::BTreeMap;
+    use crate::{AccountId, snapshot::AccountInfo, Balance, ProxyLimit, snapshot::{H256, EnclaveSnapshot}, BoundedVec, AssetsLimit, WithdrawalLimit, withdrawal::Withdrawal, assets::AssetId};
+    use frame_support::bounded_vec;
+
+    #[test]
+    fn snpashot_serialize_deserialize(){
```
Comment (serhii-temchenko) [inline src/snapshot.rs]: Typo

File: src/snapshot.rs [hunk 2]
```diff
@@ -63,4 +64,39 @@ impl<Account: std::cmp::Ord, Balance: Zero + Clone, WithdrawalLimit: Get<u32>, A
...
+            asset: AssetId::polkadex
+        };
+        withdrawals.insert(account1, bounded_vec![withdrawal.clone(), withdrawal.clone()]);
+        withdrawals.insert(account2, bounded_vec![withdrawal.clone()]);
+
+
+        let snapshot = EnclaveSnapshot::<AccountId, Balance, WithdrawalLimit, AssetsLimit>{
+            snapshot_number: 0,
+            merkle_root: H256([0_u8;32]),
+            withdrawals: withdrawals,
```
Comment (serhii-temchenko) [inline src/snapshot.rs]: Can be simplified to `withdrawals`.

File: src/snapshot.rs [hunk 3]
```diff
@@ -63,4 +64,39 @@ impl<Account: std::cmp::Ord, Balance: Zero + Clone, WithdrawalLimit: Get<u32>, A
...
+    use std::convert::From;
+    use std::collections::BTreeMap;
+    use crate::{AccountId, snapshot::AccountInfo, Balance, ProxyLimit, snapshot::{H256, EnclaveSnapshot}, BoundedVec, AssetsLimit, WithdrawalLimit, withdrawal::Withdrawal, assets::AssetId};
+    use frame_support::bounded_vec;
+
+    #[test]
+    fn snpashot_serialize_deserialize(){
+        let account1: AccountId32 = [0_u8;32].into();
+        let account2: AccountId32 = [2_u8;32].into();
+        let mut snapshot: BTreeMap<AccountId, AccountInfo<AccountId, Balance, ProxyLimit>> = Default::default();
```
Comment (serhii-temchenko) [inline src/snapshot.rs]: Not used.

File: src/snapshot.rs [hunk 4]
```diff
@@ -51,15 +53,50 @@ pub struct EnclaveSnapshot<
...
+            snapshot_number: 0,
+            merkle_root: H256([0_u8;32]),
+            withdrawals: withdrawals,
+            fees: bounded_vec![]
+        };
+        let serialized = serde_json::to_string(&snapshot).unwrap();
+        let deserialized: EnclaveSnapshot<AccountId, Balance, WithdrawalLimit, AssetsLimit> = serde_json::from_str(&serialized).unwrap();
+        assert_eq!(deserialized, snapshot);
+    }
+}
```
Comment (serhii-temchenko) [inline src/snapshot.rs]: Please add empty line.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: src/snapshot.rs (hunk 1 above)
Comment (felixfaisal) [reply src/snapshot.rs]: Updated ✅

File: src/snapshot.rs (hunk 2 above)
Comment (felixfaisal) [reply src/snapshot.rs]: Updated ✅

File: src/snapshot.rs (hunk 3 above)
Comment (felixfaisal) [reply src/snapshot.rs]: Updated ✅

File: src/snapshot.rs (hunk 4 above)
Comment (felixfaisal) [reply src/snapshot.rs]: Updated ✅

Comment (serhii-temchenko) [reply src/snapshot.rs]: Wrong line. Please add at the end.

**Review by serhii-temchenko — APPROVED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Comment by Gauthamastro** [comment]
There should be a equivalent update to 0.9.19 branch too. @felixfaisal

**Pull request merged**